### PR TITLE
Display a snackbar when draft is restored

### LINKED ISSUE #2303: [Data collection] Show message when draft is restored
A snackbar with text: "The app was closed during data collection. Your unsaved draft has been restored."

@amegantz for copy. Note that the app can be closed by the OS when in the background to conserve resources, or the user could have force quit the app, and we have no way to determine which happened.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #2303

[draft-restored-snackbar.webm](https://github.com/user-attachments/assets/83f04f6c-d141-4bfd-86ed-5f181c367c62)

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -187,4 +187,5 @@
     ## Public data sharing
     \nSurvey organizers may share and use data publicly under the *Creative Commons CC0 1.0 License*:
   </string>
+  <string name="draft_restored">App was closed. Unsaved data restored.</string>
```
Comment (gino-m) [inline ground/src/main/res/values/strings.xml:190]: @amegantz For approval

**Review by gino-m — CHANGES_REQUESTED** [review]
LGTM! @amegantz for UXW approval

**Comment by amegantz** [comment]
"Unsynced data restored"?

**Comment by gino-m** [comment]
> "Unsynced data restored"?

It's not that the data wasn't synced, it simply wasn't saved locally yet.

**Comment by amegantz** [comment]
Sounds like "Unsaved data restored" should work?

**Comment by shobhitagarwal1612** [comment]
Updated the text string. @gino-m PTAL

**Comment by shobhitagarwal1612** [comment]
@gino-m Gentle ping!

**Review by gino-m — APPROVED** [review]

**Pull request merged**